presto-native-execution/presto_cpp/main/connectors/tpcds/TpcdsGen.cpp

```diff
@@ -0,0 +1,975 @@
+    child->resize(rowCount);
+  }
+
+  return std::make_shared<RowVector>(
+      pool,
+      rowType,
+      BufferPtr(nullptr),
+      rowCount,
+      std::move(tableDef[tableId]->children));
+}
+
+RowVectorPtr genTpcdsTableData(
+    Table table,
+    memory::MemoryPool* pool,
+    size_t maxRows,
+    size_t offset,
+    double scaleFactor,
+    vector_size_t parallel,
+    vector_size_t child) {
+  //  Create schema and allocate vectors.
+  auto rowType = getTableSchema(table);
+  auto table_id = static_cast<int>(table);
+  DSDGenIterator dsdGenIterator(scaleFactor, parallel, child);
+  size_t vectorSize =
+      getVectorSize(dsdGenIterator.getRowCount(table_id), maxRows, offset);
+  auto children = allocateVectors(rowType, vectorSize, pool);
+  dsdGenIterator.initTableOffset(table_id, offset);
+  dsdGenIterator.initializeTable(children, table_id);
+  auto& tableDef = dsdGenIterator.getTableDefs();
+  for (size_t i = 0; i < vectorSize; ++i) {
+    dsdGenIterator.genRow(table_id, i + offset + 1);
+  }
+  return std::make_shared<RowVector>(
+      pool,
+      rowType,
+      BufferPtr(nullptr),
+      vectorSize,
+      std::move(tableDef[table_id]->children));
+}
+
+RowVectorPtr genTpcdsParentAndChildTable(
+    memory::MemoryPool* pool,
+    size_t maxRows,
+    size_t offset,
+    double scaleFactor,
+    vector_size_t parallel,
+    vector_size_t child,
+    Table parentTable,
+    Table childTable,
+    bool generateChildTable) {
+  // Whenever a call to generate a table which is either marked as a
+  // parent or a child is requested, both the child and parent tables
+  // need to be populated.
+
+  auto parentTableRowType = getTableSchema(parentTable);
+  DSDGenIterator dsdGenIterator(scaleFactor, parallel, child);
+  size_t parentTableVectorSize = getVectorSize(
+      dsdGenIterator.getRowCount(static_cast<int>(parentTable)),
+      maxRows,
+      offset);
+
+  size_t parentTableUpperBound = parentTableVectorSize * 16;
+  auto children =
+      allocateVectors(parentTableRowType, parentTableUpperBound, pool);
+
+  auto childTableRowType = getTableSchema(childTable);
+  auto childChildren =
+      allocateVectors(childTableRowType, parentTableUpperBound, pool);
+
+  auto parentTableId = static_cast<int>(parentTable);
+  auto childTableId = static_cast<int>(childTable);
+
+  dsdGenIterator.initTableOffset(parentTableId, offset);
+  dsdGenIterator.initializeTable(children, parentTableId);
+  dsdGenIterator.initializeTable(childChildren, childTableId);
+  auto& tableDef = dsdGenIterator.getTableDefs();
+  for (size_t i = 0; i < parentTableVectorSize; ++i) {
+    dsdGenIterator.genRow(parentTableId, i + offset + 1);
+  }
+  if (generateChildTable) {
+    return genTpcdsRowVector(childTableId, pool, childTableRowType, tableDef);
+  } else {
+    return genTpcdsRowVector(parentTableId, pool, parentTableRowType, tableDef);
+  }
+}
+} // namespace
+
+static RowTypePtr callCenterRowType = ROW(
```

Comment:
Nit : "Row" is not needed in all the types. Can be just callCenterType, catalogPageType etc.